### DISCUSSION TITLE
There should be an out of the box "power save" mode for laptops

### DISCUSSION THREAD

**Discussion by eisbehr** [linuxmint/discussions#80] [Ideas]
Hi, 
I just recently installed Linux Mint on my laptop after a few years away from desktop Linux. Over all I am quite impressed, things have improved a lot "while I was away". 
The tyranny of improvement is that it makes the smaller issues stand out more, one issue like that which would prevent me from giving an unqualified recommendation of Linux Mint to "less technical" laptop users is the seeming lack of a built in way to choose a cpu power profile. There's a number of applets and utilities to achieve it, most of which failed to work for me. I did end up finding a solution that works for me after an hour or two of fiddling. 
I think this is a generally expected feature for any laptop user and as such should be among the built in "power" settings, next to screen brightness and the like, and there should be a setting to automatically switch between "regular" and "powersave" when running from battery or wall power.

A few years ago I would have just accepted this as "the cost of running Linux", but the out of the box experience of Linux Mint has been otherwise so seamless that it stood out as *the one thing* that prevented me from having an install that "just works" on the level I would expect from a commercial operating system.

**Comment by Secret-chest** [comment]
A toggle in the battery applet for performance and powersave would be good. But these also exist on desktops, so the applet could be shown even without batteries, only displaying a lightning bolt, speedometer icon or whatever.

**Comment by rcalixte** [comment]
@eisbehr Have you had a chance to take a look at the [Power Profiles](https://cinnamon-spices.linuxmint.com/applets/view/369) applet? While it isn't a built-in applet, it is available using the `Applets` application, available under System Settings or the Main Menu.

You'd also mentioned the ones you tried failed to work. Is there any additional information or feedback you can provide here?

**Reply by eisbehr** [reply]
Power Profiles was among the ones I tried. It let me select/toggle the different profiles in the gui, but in my testing it did not appear to make a difference, like the selection didn't "take".
I also tested [CPU Frequency Applet](https://cinnamon-spices.linuxmint.com/applets/view/70), which showed the profiles for selection, but selecting them did not work (default one would always stay selected)
I then settled on a program called `indicator-cpufreq` available from the repos which works for me.

I did a run time test with the (default) schedutil governor and one with the Powersave governor as set by `indicator-cpufreq`. The second one got me significantly longer run time on battery, essentially the same as I was used to running Windows.
I also watched the CPU frequency displayed by the CPU frequency applet, which stayed noticeably low while on Powersave.
I did not do a full run time test to empty battery with the Power Profiles applet, but frequencies did not seem lower on the powersave profile and displayed remaining battery time was also consistent with "no change" after running for a few minutes on battery.

I think I found the `Power Profiles` applet last (I was searching for "cpu" and "freq", not "power" or "profile"), maybe by the time I got around to it one of the other solutions I installed conflicted with it? I didn't feel like doing much more testing after I had found one program that worked for me.

**Reply by rcalixte** [reply]
> maybe by the time I got around to it one of the other solutions I installed conflicted with it?

This is precisely one of my concerns. These applications are known to conflict with each other, impeding the intended operations of each when they do. If you're still interested in pursuing adding this functionality to your machine(s), you should make sure to be familiar with ensuring the optimal configuration occurs for whatever option you pursue.


Have you also checked to see what is compatible with your hardware? Software like power-profiles-daemon expanded functionality for AMD cards but only with 0.13 and newer releases. This is also true for the kernel and certain hardware manufacturers.

There are unfortunately too many factors at play here to be able to make much use of the information. I do hope you have a chance to properly evaluate some of the options available.

**Reply by eisbehr** [reply]
Sort of making my point for why I think it should be built-in.

**Reply by rcalixte** [reply]
> Sort of making my point for why I think it should be built-in.

I don't understand. Can you elaborate?

And which CPU governor would be the default? What if someone wants what isn't the chosen default?


The whole point of Linux is that it's customizable. The more you know about your hardware and its capabilities, the more you know how to unlock it.

**Comment by kacperpaczos** [comment]
In KDE 6, to Control power mode, they are using meta+B to switching.
https://pointieststick.com/2023/07/21/this-week-in-kde-plasma-6-features/

**Comment by anaximeno** [comment]
I believe this is something worth having out of the box too, not necessarily a CPU governor like indicator-cpufreq (the default governor thing isn't a problem as you can always make it configurable so the user can choose the defaults), but it can be used to optimize the system according to the battery level, while also allowing users to configure it manually. Most of the other major systems (GNOME, KDE, and even Windows) have one by default, and we could bake into it for some cases things like the one suggested in this discussion  https://github.com/orgs/linuxmint/discussions/282
